src/Infrastructure/Runnable.php

```diff
@@ -0,0 +1,20 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure;
+
+/**
+ * Something that can be run e.g. on a specific schedule.
+ *
+ * @since x.x.x
+ */
+interface Runnable {
```

Comment:
> I wonder if we can also define these as extensions of our current async job system. The current jobs are each doing one specific task but we could have some generic jobs that run on specific intervals and call them CRON jobs maybe?

I did consider whether this could be built on the existing job system but it looked to me there would need to be some changes required. E.g. it looked like `RecurringJobInterface` could be perfect to implement for a more generic `daily` cron job but I think that interface would need to not extend `StartOnHookInterface` or `ActionSchedulerJobInterface`. What do you think?

> If we are to go ahead and define a new set of CRON classes and interfaces, wouldn't it be better to have a separate namespace just for the CRONs? That way we can group all related services in one place.

Yes this might be better. As you mention in [your other comment](https://github.com/woocommerce/google-listings-and-ads/pull/1021#discussion_r716497567) if notes weren't responsible for adding themselves then it would make sense for a specific CRON service class to be created and have that responsibility.

Reply:
> it looked like RecurringJobInterface could be perfect to implement for a more generic daily cron job but I think that interface would need to not extend StartOnHookInterface or ActionSchedulerJobInterface. What do you think?

Hmm. Yeah, that seems like a good candidate. The jobs implementing `RecurringJobInterface` are already scheduled to run at 3 am every day (store time) and the start hook is used to schedule the cron. We probably need to refactor that interface so that it forces each job to provide an interval instead of running them all daily and on a specific date/time.

The `ResubmitExpiringProducts` job is an example of these recurring jobs that run every day. The interval and cron timing are currently set in `Jobs\JobInitializer::register`.